STM32/Libraries/SPI/Src/ADS1120.c

```diff
@@ -121,85 +140,165 @@ static HAL_StatusTypeDef writeRegister(ADS1120Device *dev, const ADS1120_RegConf
     return HAL_SPI_Transmit(dev->hspi, spiReq, 1 + count, ADS_TIMEOUT);
 }
 
-// Public functions begin.
-int ADS1120Configure(ADS1120Device *ads1120)
+static int setInput(ADS1120Device *dev, ADS1120_input selectedInput, bool verify)
 {
-    int ret;
-    const ADS1120_RegConfig cfg =
+    // A default configuration
+    ADS1120_RegConfig cfg =
     {
-            .pga_bypass = 0, // Gain should be used.
-            .gain       = 1,
-            .mux        = 0, // 0="AIN P = AIN0, AIN N = AIN1", 5="AIN P = AIN2, AIN N = AIN3"
-            .bcs        = 0, // TBD, enable Burn-out current sensor.
+            .pga_bypass = 0, // large gain should be used so this is invalid
+            .gain       = 6, // Multiply input by 32.
+            .mux        = 0, // This is altered below.
+            .bcs        = 0, // Burn-out current sensor not used
 
-            .ts         = 0, // Temperature mode
+            .ts         = 0, // Set to one if internal temp should be measured.
             .cm         = 0, // Single shot
             .mode       = 0, // Normal Mode
-            .dr         = 4, // 90 SPS
+            .dr         = 0, // 20HZ, see not in filter below.
 
-            .idac       = 0, // Must be zero, else measure is wrong.
+            .idac       = 0, // 0, since external voltage supply is used.
             .psw        = 0, // Our application is not a power bridge.
-            .fir_filter = 0, // No filter, only used for 20 SPS and 5SPS
-            .vref       = 2, // External reference selected using AIN0/REFP1 and AIN3/REFN1 inputs
+            .fir_filter = 1, // Filter on, used for 20 SPS and 5SPS and we are running at 20 HZ
+            .vref       = 0, // use internal Analog supply.
 
             .nop        = 0,
-            .drdym      = 0, // Only the dedicated DRDY pin is used to indicate when data are ready
-            .i1mux      = 0, // not routed
-            .i2mux      = 0, // not routed
+            .drdym      = 0, // The dedicated DRDY pin is used to indicate when data are ready
+            .i1mux      = 0, // not routed, since idac is 0
+            .i2mux      = 0, // not routed, since idac is 0
     };
 
-    // Select device.
-    stmSetGpio(ads1120->cs, false);
+    // Alter config to match the selected type.
+    switch(selectedInput)
+    {
+    case INPUT_TEMP:
```

Comment:
Hvordan kan det være at cfg.ts og .mux ikke sættes i alle CASE tilfældene? 
Er der risiko for at de har en forkert tilstand nogen steder?

Reply:
Fordi den struct ovenover er default. Det er så kun ændringer der laves til default i hver case setting.